src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/Compiler/AddConstraintValidatorsPassTest.php

```diff
@@ -13,56 +13,34 @@
+use Symfony\Component\DependencyInjection\Argument\ServiceLocatorArgument;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
 
 class AddConstraintValidatorsPassTest extends TestCase
 {
     public function testThatConstraintValidatorServicesAreProcessed()
     {
-        $services = array(
-            'my_constraint_validator_service1' => array(0 => array('alias' => 'my_constraint_validator_alias1')),
-            'my_constraint_validator_service2' => array(),
-        );
-
-        $validatorFactoryDefinition = $this->getMockBuilder('Symfony\Component\DependencyInjection\Definition')->getMock();
-        $container = $this->getMockBuilder('Symfony\Component\DependencyInjection\ContainerBuilder')->setMethods(array('findTaggedServiceIds', 'getDefinition', 'hasDefinition'))->getMock();
-
-        $validatorDefinition1 = $this->getMockBuilder('Symfony\Component\DependencyInjection\Definition')->setMethods(array('getClass'))->getMock();
-        $validatorDefinition2 = $this->getMockBuilder('Symfony\Component\DependencyInjection\Definition')->setMethods(array('getClass'))->getMock();
-
-        $validatorDefinition1->expects($this->atLeastOnce())
-            ->method('getClass')
-            ->willReturn('My\Fully\Qualified\Class\Named\Validator1');
-        $validatorDefinition2->expects($this->atLeastOnce())
-            ->method('getClass')
-            ->willReturn('My\Fully\Qualified\Class\Named\Validator2');
-
-        $container->expects($this->any())
-            ->method('getDefinition')
-            ->with($this->anything())
-            ->will($this->returnValueMap(array(
-                array('my_constraint_validator_service1', $validatorDefinition1),
-                array('my_constraint_validator_service2', $validatorDefinition2),
-                array('validator.validator_factory', $validatorFactoryDefinition),
-            )));
-
-        $container->expects($this->atLeastOnce())
-            ->method('findTaggedServiceIds')
-            ->will($this->returnValue($services));
-        $container->expects($this->atLeastOnce())
-            ->method('hasDefinition')
-            ->with('validator.validator_factory')
-            ->will($this->returnValue(true));
-
-        $validatorFactoryDefinition->expects($this->once())
-            ->method('replaceArgument')
-            ->with(1, array(
-                'My\Fully\Qualified\Class\Named\Validator1' => 'my_constraint_validator_service1',
-                'my_constraint_validator_alias1' => 'my_constraint_validator_service1',
-                'My\Fully\Qualified\Class\Named\Validator2' => 'my_constraint_validator_service2',
-            ));
+        $container = new ContainerBuilder();
+        $validatorFactory = $container->register('validator.validator_factory')
+            ->setArguments(array(new ServiceLocatorArgument()));
```

Comment:
broken here, the constructor expects an argument. See https://github.com/symfony/symfony/pull/21821